pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -264,22 +259,19 @@ const SourceRow = ({ connectionName, source, sourceType, networks, nodeDevices,
                     {getPXENetworkRows(nodeDevices, networks)}
                 </CockpitSelect.StatelessSelect>
 
-                {installationSourceWarning &&
-                <HelpBlock>
-                    <p className="text-warning">{installationSourceWarning}</p>
-                </HelpBlock> }
+                {installationSourceWarning && <p className="text-warning">{installationSourceWarning}</p>}
             </>
         );
         break;
     case URL_SOURCE:
         installationSourceId = "source-url";
         installationSource = (
-            <input id={installationSourceId} className="form-control"
-                type="text"
-                minLength={1}
-                placeholder={_("Remote URL")}
-                value={source}
-                onChange={e => onValueChanged('source', e.target.value)} />
+            <TextInput id={installationSourceId}
+                       validated={validationStateSource}
```

Comment:
Perhaps you want this value to also contain "default" as default value instead of undefined (since you change it everywhere else). But I don't really mind.